slsim/Deflectors/cluster_catalog_lens.py

```diff
@@ -0,0 +1,281 @@
+            or "e2_mass" not in cluster_column_names
+        ):
+            cluster_list["e1_mass"] = -np.ones(n_clusters)
+            cluster_list["e2_mass"] = -np.ones(n_clusters)
+
+        # members
+        if "z" not in members_list.columns:
+            members_list["z"] = -np.ones(len(members_list))
+            # assign the redshift of the cluster to its members
+            for i in range(n_clusters):
+                z = cluster_list["z"][i]
+                members_list["z"][
+                    members_list["cluster_id"] == cluster_list["cluster_id"][i]
+                ] = z
+        # assign a similar SLSim galaxy to each member
+        members_list = self.assign_similar_galaxy(
+            members_list, galaxy_list, cosmo=cosmo
+        )
+        n_members = len(members_list)
+        members_column_names = members_list.colnames
+        if "vel_disp" not in members_column_names:
+            members_list["vel_disp"] = -np.ones(n_members)
+        if (
+            "e1_light" not in members_column_names
+            or "e2_light" not in members_column_names
+        ):
+            members_list["e1_light"] = -np.ones(n_members)
+            members_list["e2_light"] = -np.ones(n_members)
+        if (
+            "e1_mass" not in members_column_names
+            or "e2_mass" not in members_column_names
+        ):
+            members_list["e1_mass"] = -np.ones(n_members)
+            members_list["e2_mass"] = -np.ones(n_members)
+        if "n_sersic" not in members_column_names:
+            members_list["n_sersic"] = -np.ones(n_members)
+        if "gamma_pl" not in members_column_names:
+            members_list["gamma_pl"] = np.ones(n_members) * 2
+
+        self._f_vel_disp = vel_disp_abundance_matching(
+            members_list, z_max=0.5, sky_area=sky_area, cosmo=cosmo
+        )
+
+        self._members_select = object_cut(members_list, **kwargs_cut)
+        self._cluster_select = cluster_list[
+            np.isin(cluster_list["cluster_id"], self._members_select["cluster_id"])
+        ]
+
+        self._members_select["vel_disp"] = self._f_vel_disp(
+            np.log10(self._members_select["stellar_mass"])
+        )
+
+        self._kwargs_mass2light = kwargs_mass2light
+
+        self._num_select = len(self._cluster_select)
+
+        self._cosmo = cosmo
+
+        # TODO: random reshuffle of matched list
+
+    def deflector_number(self):
+        """
+
+        :return: number of deflectors
+        """
+        number = self._num_select
+        return number
+
+    def draw_deflector(self):
+        """
+        :return: dictionary of complete parameterization of deflector
+        """
+        index = random.randint(0, self._num_select - 1)
+        deflector = self.draw_cluster(index)
+        members = self.draw_members(deflector["cluster_id"])
+        deflector["subhalos"] = members
+        return deflector
+
+    def draw_cluster(self, index):
+        cluster = self._cluster_select[index]
```

Comment:
Missing documentation for the input parameter.

Reply:
added documentation